pthread/pthread.c

```diff
@@ -483,44 +583,138 @@ int pthread_setschedparam(pthread_t thread, int policy,
 int pthread_mutex_init(pthread_mutex_t *mutex, const pthread_mutexattr_t *attr)
 {
 	(void)attr;
-
-	return mutexCreate(mutex);
+	mutex->initialized = 1;
```

Comment:
We should return -EINVAL on existing mutex

Reply:
I don't think we can check `mutex->initialized` here because it might not have been initialized at all - `pthread_mutex_init` should be called only after `pthread_mutex_t` variable declaration, without assigning it `PTHREAD_MUTEX_INITIALIZER`

Reply:
Ok, so it would be just an error from user and we can't really do anything about it (as mutex can have random data if it was created on stack)